### ISSUE TITLE
foutmelding bij het open van content op VRT NU

### ISSUE THREAD

**Issue by ErwinBaeyens** [add-ons/plugin.video.vrt.nu#939]
### Describe the bug ###
Foutmelding bij het openen van verschillende programmas
"One or more items failed to play. Check the log for more informatation about this message."

### To Reproduce ###
1. Ga naar VRT NU/ categorie / series/ twee zomers/ of Axel Nort
2. open een aflevering.

### Expected behavior ###
De aflevering of het programma start met spelen

### Additional context ###

- Operating system:
 NAME="LibreELEC"
VERSION="9.2.8"
ID="libreelec"
VERSION_ID="9.2"
PRETTY_NAME="LibreELEC (official): 9.2.8"
HOME_URL="https://libreelec.tv"
BUG_REPORT_URL="https://github.com/LibreELEC/LibreELEC.tv"
BUILD_ID="2282c381ee597a3ca6a6cf3836b5681049f7b81b"
OPENELEC_ARCH="RPi4.arm"
LIBREELEC_ARCH="RPi4.arm"
LIBREELEC_BUILD="official"
LIBREELEC_PROJECT="RPi"
LIBREELEC_DEVICE="RPi4"
RPi4.arm-9.2.8
- Kodi version: 18.9.0 Git: leia_p4_18.9-leia compiled 2021/07/06
- Addon version: 2.5.7
- Using a VPN: No
- Country you are using the addon from: Belgie

### 2022-03-13 10:52:31.533 T:2328093504   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.UnicodeDecodeError'>
                                            Error Contents: 'ascii' codec can't decode byte 0xe2 in position 58437: ordinal not in range(128)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py", line 14, in <module>
                                                run(argv)
                                              File "/storage/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon.py", line 362, in run
                                                plugin.run(argv)
                                              File "/storage/.kodi/addons/script.module.routing/lib/routing.py", line 130, in run
                                                self._dispatch(self.path)
                                              File "/storage/.kodi/addons/script.module.routing/lib/routing.py", line 141, in _dispatch
                                                view_func(**kwargs)
                                              File "/storage/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon.py", line 287, in play_id
                                                VRTPlayer().play(dict(video_id=video_id, publication_id=publication_id))
                                              File "/storage/.kodi/addons/plugin.video.vrt.nu/resources/lib/vrtplayer.py", line 400, in play
                                                play(stream, video.get('listitem'))
                                              File "/storage/.kodi/addons/plugin.video.vrt.nu/resources/lib/kodiutils.py", line 315, in play
                                                if is_helper.check_inputstream():
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/__init__.py", line 407, in check_inputstream
                                                return self._check_drm()
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/__init__.py", line 359, in _check_drm
                                                return self._check_widevine()
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/__init__.py", line 329, in _check_widevine
                                                self._update_widevine()
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/__init__.py", line 288, in _update_widevine
                                                latest_version = latest_widevine_version()
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/widevine/widevine.py", line 164, in latest_widevine_version
                                                devices = chromeos_config()
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/widevine/arm.py", line 63, in chromeos_config
                                                return json.loads(http_get(config.CHROMEOS_RECOVERY_URL))
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/utils.py", line 73, in http_get
                                                return content.decode()
                                            UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 58437: ordinal not in range(128)
                                            -->End of Python script error report<--
2022-03-13 10:52:31.668 T:3011052928   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.vrt.nu/play/id/vid-342a88ed-7e63-4106-9fe8-e8ebb51474ae/pbs-pub-7d58d04d-5d24-4107-8bd7-3c1ef7795be5]
2022-03-13 10:53:20.843 T:2354574144   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.UnicodeDecodeError'>
                                            Error Contents: 'ascii' codec can't decode byte 0xe2 in position 58437: ordinal not in range(128)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py", line 14, in <module>
                                                run(argv)
                                              File "/storage/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon.py", line 362, in run
                                                plugin.run(argv)
                                              File "/storage/.kodi/addons/script.module.routing/lib/routing.py", line 130, in run
                                                self._dispatch(self.path)
                                              File "/storage/.kodi/addons/script.module.routing/lib/routing.py", line 141, in _dispatch
                                                view_func(**kwargs)
                                              File "/storage/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon.py", line 287, in play_id
                                                VRTPlayer().play(dict(video_id=video_id, publication_id=publication_id))
                                              File "/storage/.kodi/addons/plugin.video.vrt.nu/resources/lib/vrtplayer.py", line 400, in play
                                                play(stream, video.get('listitem'))
                                              File "/storage/.kodi/addons/plugin.video.vrt.nu/resources/lib/kodiutils.py", line 315, in play
                                                if is_helper.check_inputstream():
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/__init__.py", line 407, in check_inputstream
                                                return self._check_drm()
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/__init__.py", line 359, in _check_drm
                                                return self._check_widevine()
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/__init__.py", line 329, in _check_widevine
                                                self._update_widevine()
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/__init__.py", line 288, in _update_widevine
                                                latest_version = latest_widevine_version()
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/widevine/widevine.py", line 164, in latest_widevine_version
                                                devices = chromeos_config()
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/widevine/arm.py", line 63, in chromeos_config
                                                return json.loads(http_get(config.CHROMEOS_RECOVERY_URL))
                                              File "/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/utils.py", line 73, in http_get
                                                return content.decode()
                                            UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 58437: ordinal not in range(128)
                                            -->End of Python script error report<--
2022-03-13 10:53:21.007 T:3011052928   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.vrt.nu/play/id/vid-ca19e8f6-1c81-42a0-ad9d-0dbfe8f3d54a/pbs-pub-7eae582b-bd6c-4972-af9c-1e8dd7b78a60]
Log (if available) ###

```

**Comment by tebokial** [comment]
Hello, 
1.  I  encountered the same "ascii decoding" error yesterday and today, both playing "Live TV" and "Recent", running CoreElec 9.2.8 (a cousin of Kodi 18.x on LibreElec) on an OdroidN2plus. But before I got there, I got into trouble updating the  widevine CDM : nothing seemed to work (I went into the Inputstream adaptive and tried "remove and re-install").  
[kodi_odroidn2_Leia18_9.log](https://github.com/add-ons/plugin.video.vrt.nu/files/8239907/kodi_odroidn2_Leia18_9.log)
2.  Just to exclude that this is not an "Old Leia" vs. "New Matrix" issue, I then installed Kodi Matrix 19.4, again using CoreElec.  This time, when trying to play Live TV in VRT Nu, I succeeded in completing the install of the Widevine CDM, but then ... just a lot of buffering and a black screen .... I am sure that Widevine is again to blame.   I am seriously thinking of jumping the "ARM ship" as my Kodi platform ... but wait, weren't we all supposed to use energy efficient devices, like Raspberry Pi's and Odroid ?    Shame on VRT, our "public" broadcast (e.g. the German public broadcasters have not embraced this proprietary DRM madness) ... sorry for the rant.

**Comment by Roel** [comment]
The issue is not specific to VRT Nu, but is a general issue in inputstreamhelper regarding the Widevine version check: https://github.com/emilsvennesson/script.module.inputstreamhelper/issues/495

The fix is discussed in the Github issue and from the looks of things will be available in a bugfix release in the next few days.

**Comment by ErwinBaeyens** [comment]
I added the proposed solution to that file from inputstreamhelper and it works for me:
https://github.com/emilsvennesson/script.module.inputstreamhelper/issues/495

**Comment by doodhout** [comment]
The solution, summarised:

> Changing line 73 in lib/inputstreamhelpers/util.py to the following seems to work for me:
return content.decode("utf-8")
>
> Seems that the file used to get the current version of Widevine has been changed on googles side...

(source: https://github.com/emilsvennesson/script.module.inputstreamhelper/issues/495#issuecomment-1065477688)

> For anyone in doubt: the correct file name is utils.py (notice the s). In case someone like me uses LibreElec (Kodi appliance type OS) the full path to the file in question is /storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/utils.py
> so one can login via ssh to their kodi box and edit it with
>
> `nano /storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/utils.p`

(source: https://github.com/emilsvennesson/script.module.inputstreamhelper/issues/495#issuecomment-1065978855)

> SSH to your raspberry pi e. g. using Putty: `ssh root@libreelec` with password libreelec unless you changed the default password. Then call `nano /storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/utils.py` as mentioned above and navigate to line 73. Change the line, press ctrl + o and ctrl+ x so save and exit the file and restart you pi.

(source: https://github.com/emilsvennesson/script.module.inputstreamhelper/issues/495#issuecomment-1066822860)

TL;DR: 

1. `nano /storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper/utils.p`
2. change line 73 by adding `"utf-8"` between the brackets (or _parentheses_ for the peeps who speak americano)
3. save the file by pressing CTRL+W to write the changes to disk and CTRL+X to exit the editor
4. start your favourite VRTNU stream, perhaps dismissing the suggestion to update Widevine if you suffer(ed) from this bug: https://github.com/add-ons/plugin.video.vrt.nu/issues/933

**Comment by ythefoe** [comment]
Thanks, that fixed it for me!

**Comment by mediaminister** [comment]
A new version of InputStreamHelper that fixes this problem has been released.
